core/util.py

```diff
@@ -0,0 +1,15 @@
+from core.const import POSTCODE_TO_ITL1
+
+
+def postcode_to_itl1(postcode: str) -> str:
+    """
+    Maps a given UK postcode to its corresponding ITL1 code.
+
+    :param postcode: A string representing a UK postcode.
+    :return: A string representing the corresponding ITL1 code.
+    :raises IndexError: If the given postcode is invalid and has no mapping.
+    """
+    try:
+        return POSTCODE_TO_ITL1[postcode]
```

Comment:
We should be looking for a match at the start of the supplied full postcode. e.g. an input postcode like `W1A 1AA` should return `TLI` matching should also be case-insensitive.

Reply:
Good point 👍

Reply:
We probably should trim whitespace too to allow for malformed data

Reply:
This has been implemented now